jms/org.wso2.carbon.transport.jms/src/test/java/org/wso2/carbon/transport/jms/test/QueueTopicAutoAckListeningTestCase.java

```diff
@@ -0,0 +1,126 @@
+                .put(JMSConstants.CONNECTION_FACTORY_JNDI_PARAM_NAME, JMSTestConstants.QUEUE_CONNECTION_FACTORY);
+        queueListeningParameters
+                .put(JMSConstants.NAMING_FACTORY_INITIAL_PARAM_NAME, JMSTestConstants.ACTIVEMQ_FACTORY_INITIAL);
+        queueListeningParameters.put(JMSConstants.PROVIDER_URL_PARAM_NAME, JMSTestConstants.ACTIVEMQ_PROVIDER_URL);
+        queueListeningParameters
+                .put(JMSConstants.CONNECTION_FACTORY_TYPE_PARAM_NAME, JMSConstants.DESTINATION_TYPE_QUEUE);
+
+        topicListeningParameters = new HashMap<>();
+        topicListeningParameters.put(JMSConstants.DESTINATION_PARAM_NAME, JMSTestConstants.TOPIC_NAME);
+        topicListeningParameters
+                .put(JMSConstants.CONNECTION_FACTORY_JNDI_PARAM_NAME, JMSTestConstants.TOPIC_CONNECTION_FACTORY);
+        topicListeningParameters
+                .put(JMSConstants.NAMING_FACTORY_INITIAL_PARAM_NAME, JMSTestConstants.ACTIVEMQ_FACTORY_INITIAL);
+        topicListeningParameters.put(JMSConstants.PROVIDER_URL_PARAM_NAME, JMSTestConstants.ACTIVEMQ_PROVIDER_URL);
+        topicListeningParameters
+                .put(JMSConstants.CONNECTION_FACTORY_TYPE_PARAM_NAME, JMSConstants.DESTINATION_TYPE_TOPIC);
+        jmsServer = new JMSServer();
+        jmsServer.startServer();
+
+        // Create a queue transport listener
+        jmsQueueTransportListener = new JMSServerConnector("1");
+        queueTestMessageProcessor = new TestMessageProcessor();
+        jmsQueueTransportListener.setMessageProcessor(queueTestMessageProcessor);
+
+        // Create a topic transport listener
+        jmsTopicTransportListener = new JMSServerConnector("2");
+        topicTestMessageProcessor = new TestMessageProcessor();
+        jmsTopicTransportListener.setMessageProcessor(topicTestMessageProcessor);
+    }
+
+    /**
+     * JMS Server connector will start listening to a queue in auto ack mode and there will be a publisher publishing to
+     * the queue. After publishing the messages to queue, check whether the count of messages received is equal to
+     * number of messages sent.
+     */
+    @Test(groups = "jmsListening", description = "Testing whether queue listening is working correctly without any "
+            + "exceptions in auto ack mode")
+    public void queueListeningTestCase() {
+        try {
+            jmsQueueTransportListener.start(queueListeningParameters);
+            logger.info("JMS Transport Listener is starting to listen to the queue " + JMSTestConstants.QUEUE_NAME);
+            jmsServer.publishMessagesToQueue(JMSTestConstants.QUEUE_NAME);
+            Assert.assertEquals(queueTestMessageProcessor.getCount(), 10,
+                    "Expected message count is not received when " + "listening to queue "
+                            + JMSTestConstants.QUEUE_NAME);
+            jmsQueueTransportListener.stop();
+        } catch (Exception e) {
+            Assert.fail("Error while listening to queue");
```

Comment:
Simply throwing the error from the test case would fail it. We don't have to catch and Assert fail.